src/engine/idTable/IdTable.h

```diff
@@ -346,7 +365,8 @@ class IdTable {
   // Get a reference to the `i`-th row. The returned proxy objects can be
   // implicitly and trivially converted to `row_reference`. For the design
   // rationale behind those proxy types see above for their definition.
-  row_reference_restricted operator[](size_t index) requires(!isView) {
+  CPP_template(typename = void)(requires(!isView)) row_reference_restricted
```

Comment:
This pattern happens very often, maybe wriite two macros at the beginning of this fiile
`define REQUIRES_VIEW CPP_template(typename = void) (requires isView)`
And the same for `REQUIRES_NOT_VIEW`, then you can write
`REQUIRES_NOT_VIEW row_reference_restricted operator[](size_t index)` here (similar in the other places),
whiich is nicer and uses less space.
Don't forget to undefine the macros again at the end of the header file to not clutter the global namespace.

Reply:
my guess is that this will be more common that just this file, what about enhancing concepts.h with some macro like:
`CPP_parameterless_template(...) CPP_template(__VA_ARGS__)

Reply:
Something like that could work, I will look into it, as I have discovered some additional macros to be missing.

Reply:
For this PR (to get it merged) we will leave the `typename = void` defaults, 
But if I have time I will add some additional macros.